Runtime/Scripts/OscSocket.cs

```diff
@@ -74,15 +106,26 @@ void Serve()
                     break;
                 }
             }
-            
+
+            m_ThreadWakeup.Dispose();
             Profiler.EndThreadProfiling();
```

Comment:
I know this line isn't new, but I noticed that we run `Profiler.EndThreadProfiling()` everywhere, while we only run  `Profiler.BeginThreadProfiling()` in the Editor.

Reply:
Hmm, there's also a spurious `Profiler.EndSample();` there... I'll remove that and put `EndThreadProfiling` behind `#if UNITY_EDITOR`, yeah?